src/canvasdrawer.js

```diff
@@ -0,0 +1,1151 @@
+/*
+ * OpenSeadragon - Drawer
```

Comment:
Will be fixed in the next commit. Question - right below this where the years of the copyright are listed, since this is a new file do we need to include only the years when it existed? What about the codeplex year? Since code that existed for a long time was copied over into this implementation I left those in, but I'm not sure what official policy is.

Reply:
I don't know that we have an official policy, but I think just keeping the copyright block the same for every file is probably best.

Speaking of which, looks like we need to update to 2023 or maybe just go ahead and take it to 2024? That doesn't necessarily need to happen in this patch, but it could.